Testing/SystemTests/tests/framework/DocumentationTest.py

```diff
@@ -0,0 +1,161 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import pathlib
+from pathlib import Path
+from typing import List, Dict
+
+from systemtesting import MantidSystemTest
+from mantid.api import AlgorithmFactory, FunctionFactory
+from mantid.kernel import ConfigService
+from workbench.utils.gather_interfaces import gather_interface_names
+from mantidqt.interfacemanager import InterfaceManager
+
+path_to_docs = (pathlib.Path(__file__).resolve() / "../../../../../docs/source").resolve()
+ALG_DOCS_PATH = path_to_docs / "algorithms"
+FIT_DOCS_PATH = path_to_docs / "fitting/fitfunctions"
+INTERFACES_DOCS_PATH = path_to_docs / "interfaces"
+
+# This should be empty
+ALGORITHMS_WITH_NO_DOCS = [
+    "ApplyNegMuCorrection-v1.rst",
+    "DPDFreduction-v1.rst",
+    "LoadCSNSNexus-v1.rst",
+    "LoadEMUHdf-v1.rst",
+    "LoadILLLagrange-v1.rst",
+    "LoadILLSALSA-v1.rst",
+    "OrderWorkspaceHistory-v1.rst",
+    "PatchBBY-v1.rst",
+    "RefinePowderInstrumentParameters-v2.rst",
+    "ReflectometryISISPreprocess-v1.rst",
+    "ReflectometryISISSumBanks-v1.rst",
+    "SANSBeamCentreFinderCore-v1.rst",
+    "SANSBeamCentreFinderMassMethod-v1.rst",
+    "SANSBeamCentreFinder-v1.rst",
+    "SANSPatchSensitivity-v1.rst",
+    "SANSReductionCoreEventSlice-v1.rst",
+    "SANSReductionCorePreprocess-v1.rst",
+    "SANSReductionCore-v1.rst",
+    "SANSSingleReduction-v1.rst",
+    "SANSSingleReduction-v2.rst",
+]
+# This should be empty
+FIT_FUNCTIONS_WITH_NO_DOCS = [
+    "ConvTempCorrection.rst",
+    "Example1DFunction.rst",
+    "ExamplePeakFunction.rst",
+    "FlatTopPeak.rst",
+    "PawleyParameterFunction.rst",
+    "PeakParameterFunction.rst",
+    "PoldiSpectrumConstantBackground.rst",
+    "PoldiSpectrumDomainFunction.rst",
+    "PoldiSpectrumLinearBackground.rst",
+    "PoldiSpectrumPawleyFunction.rst",
+    "ReflectivityMulf.rst",
+    "SCDCalibratePanels2ObjFunc.rst",
+    "SCgapSwave.rst",
+    "VesuvioResolution.rst",
+]
+# This should be empty
+INTERFACES_WITH_NO_DOCS = [
+    "Shiver.rst",  # externally imported
+    "SNAPRed.rst",  # externally imported
+    "ORNL SANS.rst",  # information is in concepts/ORNL_SANS_Reduction.rst
+    "Powder Diffraction Reduction.rst",  # reduction application
+    "DGS Reduction.rst",  # reduction application
+    "Direct Data Analysis.rst",  # opens Inelastic Data Analysis
+    "Indirect Data Analysis.rst",  # opens Inelastic Data Analysis
```

Comment:
This interface will be moved to the `Inelastic` category in the future, which will fix this issue

Reply:
if there's an issue for that then we should leave a comment to mention this new test.